cms/djangoapps/contentstore/views/item.py

```diff
@@ -1027,10 +1031,10 @@ def _compute_visibility_state(xblock, child_info, is_unit_with_changes):
         # Note that a unit that has never been published will fall into this category,
         # as well as previously published units with draft content.
         return VisibilityState.needs_attention
+
     is_unscheduled = xblock.start == DEFAULT_START_DATE
-    is_live = datetime.now(UTC) > xblock.start
-    children = child_info and child_info.get('children', [])
-    if children and len(children) > 0:
+    is_live = is_course_self_paced or datetime.now(UTC) > xblock.start
```

Comment:
Nice catch, but I didn't wanted to pass whole course object which was unnecessary, so I passed only what is required in `_compute_visibility_state` method. 

However, I should make  `is_course_self_paced` param to be false by default.